languages/haskell/docs/implementing-a-concept-exercise.md

```diff
@@ -2,48 +2,61 @@
 
 This document describes the steps required to implement a concept exercise for the Haskell track. As this document is generic, the following placeholders are used:
 
-- `<SLUG>`: the name of the exercise in kebab-case (e.g. `anonymous-methods`).
-- `<NAME>`: the name of the exercise in PascalCase (e.g. `AnonymousMethods`).
+- `<SLUG>`: the name of the exercise in kebab-case (e.g. `monad-transformers`).
+- `<NAME>`: the name of the exercise in PascalCase (e.g. `MonadTransformers`).
 
-Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue. Concept exercises should be modelled after the CIS 194 Haskell course at Penn University, either the [Spring 2013][spring-2013] or [Fall 2016][fall-2016] version.
+Haskell covers a vast amount of concepts. Choosing a particular subset of concepts and an ideal ordering of these concepts is a contentious task, so the following approach is made: As a basis for both, the online resources for the CIS 194 course at Penn University is chosen.
 
-To implement a concept exercise, the following files need to be created:
+Contributors (you) may freely choose a concept related to Haskell and implement an exercise that covers this. Before implementing such an exercise, make sure you have a good understanding of what the exercise should be teaching (and what not). Check if this concept is already partially or fully covered by another exercise, and arrange the concept in a hierarchical fashion that corresponds to the curriculum of CIS 194.
+
+Because CIS 194 online course material is available in the [Spring 2013][spring-2013] and the [Fall 2016][fall-2016] version, and because these have different orderings, this does leave room for interpretation.
+
+Because CIS 194 may not cover all subject that contributors wish to cover in this material, extending the concepts beyond the CIS 194 course should ideally be accompanied by some level of reasoning about the curriculum.
+
+The directory structure of a single concept exercise looks like this:
 
 ```
 languages
 └── Haskell
     └── exercises
         └── concept
             └── <SLUG>
-                ├── .docs
-                |   ├── instructions.md
-                |   ├── introduction.md
-                |   ├── hints.md
-                |   └── after.md (optional)
-                ├── .meta
-                |   |── design.md
-                |   └── <SLUG>.hs (example solution)
+                ├── package.yaml
+                ├── stack.yaml
                 ├── src
-                |   └── <NAME>.hs
+                │   └── <NAME>.hs
                 ├── test
-                |   └── Tests.hs
-                ├── package.yaml
-                └── stack.yaml
+                │   └── Tests.hs
+                ├── examples
+                │   ├── success-... (default: standard)
```

Comment:
Just out of curiosity: are you expecting to implement multiple examples?

Reply:
In the context of contributing concept exercises, no.

In some future contexts of the track, yes. This PR merges the current track structure with the track structure intended for working with concept exercises. The current track structure allows for multiple examples. So a track structure that merges the current track structure would either also allow for this or disallow it for all exercises that currently have multiple examples.

If we were to remove multiple examples, one question is if we're changing it because we have revised our general opinion on the possibility for multiple examples, or if it's a practical concern that only relates to concept exercises.

I raised the issue at exercism/haskell#893 where discussion has not progressed. Until another consensus is made, I presume the path of least contention involves merging the possibility for multiple examples.

Reply:
Yep, that makes sense! Thanks for explaining.